src/httpClient.ts

```diff
@@ -134,12 +145,13 @@ class DefaultHTTPClient implements HTTPClient {
       error = null;
       request_start = performance.now();
       try {
-        response = await fetch(url.toString(), {
-          method,
+        response = await this.agent.request({
```

Comment:
Did you verify that this still pools properly? 👍🏻

Reply:
yes it definitely does